### ISSUE TITLE
Add option to use a system toolbar theme

### ISSUE THREAD

**Issue by ghostplant** [rilian-la-te/vala-panel#87]
I know that vala-panel can configure background color manually just by its plugin configuration dialog. However, I'd like it to use system default color configuration thus it's background will change together with the GTK3 theme changes.

Currently, I found that Ubuntu default Ambiance themes doesn't work well with vala-panel using system theme: Ambiance is dark theme while vala-panel under Ambiance is of light color. How to write a gtk3.0 css line to update this background color, just like the format below:

```sh
NautilusWindow, GtkDialog, GtkInspectorWindow, GeditPreferencesDialog {
	box-shadow: 0 4px 4px rgba(0, 0, 0, 0.6);
}
```

**Comment by rilian-la-te** [comment]
Apply custom CSS is not working on vala-panel anymore?

**Comment by rilian-la-te** [comment]
If not, it is a bug.

**Comment by rilian-la-te** [comment]
But CSS you posted is not working on 3.22

**Comment by rilian-la-te** [comment]
Or just use dark theme option, it should work.

**Comment by ghostplant** [comment]
Enable/disable "dark theme" checkbox from vala-panel option doesn't have any different on Ambiance/Radiance/MATE theme.

I just don't know hot to write custom CSS for vala-panel, and the fact is that even if Ambiance is a typical dark theme, the vala-panel is still light background (whatever "dark theme" checkbox is set or not), the Ambiance icon theme is mostly white decorated so light panel background and white decorated icons make them hard to distinguish.

**Comment by ghostplant** [comment]
Actually, both Ambiance and MATE themes agree that the panel system background color should be STYLE_CLASS_PRIMARY_TOOLBAR, while vala-panel doesn't, so this makes Ambiance and MATE themes doesn't keep same style with each other. Is it possible to make the vala-panel background to apply for STYLE_CLASS_PRIMARY_TOOLBAR? I only know how python-gi does:

```sh
Gtk.Window.get_style_context().add_class(Gtk.STYLE_CLASS_PRIMARY_TOOLBAR)
```

**Comment by rilian-la-te** [comment]
Ambiance have a dark variant at all? If yes, dark theme should work.
Try Totem or any GNOME dark app.

If not, I can add STYLE_CLASS_PRIMARY_TOOLBAR to vala-panel (when dark theme enabled), but I do not think than it is a right style class.

**Comment by ghostplant** [comment]
I think it has dark variant since Ambiance css resources have `@dark_bg_color`.

Ambiance dark theme set `title/toolbar` to use dark style and set `document-context` to use white style, and gnome-panel/unity/mate-panel use system toolbar theme as their background theme, while vala-panel use `document-context` as its background theme.

The theme package can be access here, which is the primary ubuntu theme: http://archive.ubuntu.com/ubuntu/pool/main/u/ubuntu-themes/ubuntu-themes_16.10+18.04.20180122.1.orig.tar.gz

**Comment by rilian-la-te** [comment]
It does not have a dark variant, I checked it. I will add an option to use toolbar theme as panel theme.

**Comment by ghostplant** [comment]
OK~

**Comment by rilian-la-te** [comment]
If you want a custom CSS, you should use apply custom CSS. It is standard gtk3 css. But do not try to insert a entire system theme to it))

**Comment by ghostplant** [comment]
Yes, I think the most easy way for users is vala-panel can use toolbar-theme which keeps same style with mate-panel/gnome-panel thus most themes developed for ubuntu will also work correctly for vala-panel.

**Comment by rilian-la-te** [comment]
How to use it. Just open yout profile config file and set ```use-toolbar-appearance=true``` to your toplevel unit section.
I will add GUI later.

**Comment by ghostplant** [comment]
The lastest master commit of the upstream branch cannot pass the cmake compiling procedure:
```sh
In file included from /home/xxxx/vala-panel/ui/applets-new/applet-info.c:1:0:
/home/xxxx/vala-panel/ui/applets-new/applet-info.h:4:10: fatal error: glib-object.h: 没有那个文件或目录(No such file or directory)
 #include <glib-object.h>
          ^~~~~~~~~~~~~~~
```
This didn't happen in old commit.

**Comment by ghostplant** [comment]
my compiling procedure is:

./configure --prefix=`pwd`/../build
make

**Comment by ghostplant** [comment]
It seems that the submodule configuration is not correct.
I do the following and it works:

1) fallback to an old commit, then updating git submodule:
2) checkout the lastest commit, then updating git submodule:
3) configure and make install (works);

While the following procedure doesn't work:
1) clone a fresh master branch:
2) checkout the lastest commit, then updating git submodule:
3) configure and make install (not work, reporting lacking libglib.h);

**Comment by ghostplant** [comment]
If I rebase your master branch and drop the following 3 commits:

d760fd506251329bceab413ed001224e9ce83c7b
9108aa07dbb9ab17b4f2271df36d9da13f4d07ad
2394dba8076908faa85ddf6ba81ff1ea0d003817

The compilation will be ok.

**Comment by ghostplant** [comment]
After dropping commits that fails the compilation, I see the toolbar-appearance truly applied on vala-panel, thanks!

**Comment by rilian-la-te** [comment]
About CMake - please post details, because compilation works for me.
Do you have your submodules updated via
```git submodule update --remote --merge```?

**Comment by rilian-la-te** [comment]
Did you tried a compile procedure with CMake, not configure?

**Comment by rilian-la-te** [comment]
If you want to build a package, you should not use configure. But I will try to use it. But most likely I will drop it.

**Comment by rilian-la-te** [comment]
I will look for this tomorrow.

**Comment by rilian-la-te** [comment]
Fixed compilation in latest master, try it)

**Comment by ghostplant** [comment]
Yes, it is fixed. I'll close this issue, thanks!

**Comment by rilian-la-te** [comment]
No, GUI required, so an issue will be opened.

**Comment by denikris** [comment]
adapta-project has themed this panel btw. maybe you can fork arc-theme (or other theme) to make it default panel theme like what budgie-desktop done for its panel, for later release

**Comment by rilian-la-te** [comment]
Why? I do not want this, I want it to look as Adwaita provided to me.
